### ISSUE TITLE
Changing restart-dependent renderer effects crashes the game on vid_restart or map load

### ISSUE THREAD

**Issue by kharnov** [Unvanquished/Unvanquished#823]
/set r_ssao 1 (or 0 if you had it off before)
/vid_restart
OR
/callvote map forlorn

```Program received signal SIGSEGV, Segmentation fault.
GLShader::BindProgram (this=0x0, deformIndex=deformIndex@entry=0) at /home/kharnov/Unvanquished/daemon/src/engine/renderer/gl_shader.cpp:1179
1179            int macroIndex = SelectProgram();
(gdb) bt
#0  GLShader::BindProgram (this=0x0, deformIndex=deformIndex@entry=0) at /home/kharnov/Unvanquished/daemon/src/engine/renderer/gl_shader.cpp:1179
#1  0x00000000004d995b in RB_RenderSSAO () at /home/kharnov/Unvanquished/daemon/src/engine/renderer/tr_backend.cpp:2945
#2  0x00000000004dccaf in RB_RenderView () at /home/kharnov/Unvanquished/daemon/src/engine/renderer/tr_backend.cpp:4485
#3  RB_DrawView (data=data@entry=0x7fffcced27e4) at /home/kharnov/Unvanquished/daemon/src/engine/renderer/tr_backend.cpp:5261
#4  0x00000000004df528 in RB_ExecuteRenderCommands (data=0x7fffcced27e4) at /home/kharnov/Unvanquished/daemon/src/engine/renderer/tr_backend.cpp:5666
#5  0x00000000004ead0b in R_IssueRenderCommands (runPerformanceCounters=runPerformanceCounters@entry=false) at /home/kharnov/Unvanquished/daemon/src/engine/renderer/tr_cmds.cpp:183
#6  0x00000000004ead5b in R_SyncRenderThread () at /home/kharnov/Unvanquished/daemon/src/engine/renderer/tr_cmds.cpp:209
#7  0x00000000004ead92 in R_SyncRenderThread () at /home/kharnov/Unvanquished/daemon/src/engine/renderer/tr_cmds.cpp:204
#8  0x00000000004f8f65 in RE_GenerateTexture (pic=0xab2aa40 "\377\377\377", width=256, height=512) at /home/kharnov/Unvanquished/daemon/src/engine/renderer/tr_image.cpp:3269
#9  0x0000000000460441 in CGameVM::<lambda(std::vector<unsigned char, std::allocator<unsigned char> >, int, int, qhandle_t&)>::operator() (__closure=<optimized out>, handle=<optimized out>, y=512,
    x=<optimized out>, data=...) at /home/kharnov/Unvanquished/daemon/src/engine/client/cl_cgame.cpp:1547
#10 Util::apply_impl<CGameVM::QVMSyscall(int, Util::Reader&, IPC::Channel&)::<lambda(std::vector<unsigned char>, int, int, qhandle_t&)>, std::tuple<std::vector<unsigned char, std::allocator<unsigned char> >&&, int&&, int&&, int&>, 0ul, 1ul, 2ul, 3ul> (func=<optimized out>, tuple=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/common/Util.h:103
#11 Util::apply<CGameVM::QVMSyscall(int, Util::Reader&, IPC::Channel&)::<lambda(std::vector<unsigned char>, int, int, qhandle_t&)>, std::tuple<std::vector<unsigned char, std::allocator<unsigned char> >&&, int&&, int&&, int&> > (func=<optimized out>, tuple=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/common/Util.h:108
#12 IPC::detail::HandleMsg<CGameVM::QVMSyscall(int, Util::Reader&, IPC::Channel&)::<lambda(std::vector<unsigned char>, int, int, qhandle_t&)>, IPC::Message<IPC::Id<0u, 71u>, std::vector<unsigned char, std::allocator<unsigned char> >, int, int>, IPC::Reply<int> > (func=<optimized out>, reader=<error reading variable: access outside bounds of object referenced via synthetic pointer>, channel=...)
    at /home/kharnov/Unvanquished/daemon/src/common/IPC/Channel.h:217
#13 IPC::HandleMsg<IPC::SyncMessage<IPC::Message<IPC::Id<0u, 71u>, std::vector<unsigned char, std::allocator<unsigned char> >, int, int>, IPC::Reply<int> >, CGameVM::QVMSyscall(int, Util::Reader&, IPC::Channel&)::<lambda(std::vector<unsigned char>, int, int, qhandle_t&)> >(IPC::Channel &, Util::Reader, <unknown type in /home/kharnov/Unvanquished/build/daemon, CU 0x26eebc, DIE 0x332089>) (channel=..., reader=...,
    func=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/common/IPC/Channel.h:241
#14 0x00000000004696ba in CGameVM::QVMSyscall (this=<optimized out>, index=71, reader=..., channel=...) at /home/kharnov/Unvanquished/daemon/src/engine/client/cl_cgame.cpp:1548
#15 0x000000000046a2b7 in CGameVM::Syscall (this=<optimized out>, id=<optimized out>, reader=..., channel=...) at /home/kharnov/Unvanquished/daemon/src/engine/client/cl_cgame.cpp:1265
#16 0x0000000000470f7d in void VM::VMBase::SendMsg<IPC::SyncMessage<IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<> >, cgClientState_t&>(cgClientState_t&)::{lambda(unsigned int, Util::Reader)#1}::operator()(unsigned int, Util::Reader) (reader=<error reading variable: access outside bounds of object referenced via synthetic pointer>, id=71, __closure=0x7fffffffd280)
    at /home/kharnov/Unvanquished/daemon/src/engine/framework/VirtualMachine.h:136
#17 IPC::detail::SendMsg<void VM::VMBase::SendMsg<IPC::SyncMessage<IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<> >, cgClientState_t&>(cgClientState_t&)::{lambda(unsigned int, Util::Reader)#1}&, IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<>, cgClientState_t&>(IPC::Channel&, void VM::VMBase::SendMsg<IPC::SyncMessage<IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<> >, cgClientState_t&>(cgClientState_t&)::{lambda(unsigned int, Util::Reader)#1}&, IPC::SyncMessage<IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<> >, cgClientState_t&) (channel=..., messageHandler=...) at /home/kharnov/Unvanquished/daemon/src/common/IPC/Channel.h:174
#18 0x000000000046dc92 in IPC::SendMsg<IPC::SyncMessage<IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<> >, void VM::VMBase::SendMsg<IPC::SyncMessage<IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<> >, cgClientState_t&>(cgClientState_t&)::{lambda(unsigned int, Util::Reader)#1}, cgClientState_t&>(IPC::Channel&, void VM::VMBase::SendMsg<IPC::SyncMessage<IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<> >, cgClientState_t&>(cgClientState_t&)::{lambda(unsigned int, Util::Reader)#1}&&, cgClientState_t&)
    (messageHandler=<unknown type in /home/kharnov/Unvanquished/build/daemon, CU 0x26eebc, DIE 0x3309b5>, channel=...) at /home/kharnov/Unvanquished/daemon/src/common/IPC/Channel.h:234
#19 VM::VMBase::SendMsg<IPC::SyncMessage<IPC::Message<IPC::Id<(unsigned short)0, (unsigned short)10>, cgClientState_t>, IPC::Reply<> >, cgClientState_t&>(cgClientState_t&) (this=0xd32720 <cgvm>)
    at /home/kharnov/Unvanquished/daemon/src/engine/framework/VirtualMachine.h:134
#20 CGameVM::CGameRocketFrame (this=0xd32720 <cgvm>) at /home/kharnov/Unvanquished/daemon/src/engine/client/cl_cgame.cpp:1252
#21 0x0000000000493178 in SCR_UpdateScreen () at /home/kharnov/Unvanquished/daemon/src/engine/client/cl_scrn.cpp:502
#22 0x000000000048c080 in CL_Frame (msec=200) at /home/kharnov/Unvanquished/daemon/src/engine/client/cl_main.cpp:3556
#23 0x0000000000424411 in Com_Frame () at /home/kharnov/Unvanquished/daemon/src/engine/qcommon/common.cpp:1911
#24 0x000000000041fb25 in main (argc=<optimized out>, argv=<optimized out>) at /home/kharnov/Unvanquished/daemon/src/engine/framework/System.cpp:670```

**Comment by Viech** [comment]
@kharnov, reason for closing?

**Comment by Kangz** [comment]
I think it was fixed by 8012e3e959c3fc34f25ab2dec024ad20760e3366
